hibike/devices/Potentiometer/potentiometer.cpp

```diff
@@ -1,59 +1,51 @@
 #include "potentiometer.h"
-#include <Servo.h>
 
-
-
-uint8_t pins[NUM_PINS] = {IN_0, IN_1, IN_2, IN_3};
+uint8_t pins[NUM_PINS] = {IN_0, IN_1, IN_2};
 
 void setup() {
   hibike_setup();
 
   // Setup sensor input
   for (int i = 0; i < NUM_PINS; i++) {
-    pinMode(pins[i], INPUT_PULLUP);
+    pinMode(pins[i], INPUT);
   }
-
 }
 
 
 void loop() {
   hibike_loop();
 }
 
-// you must implement this function. It is called when the device receives a DeviceUpdate packet.
-// the return value is the value field of the DeviceRespond packet hibike will respond with
-uint32_t device_update(uint8_t param, uint32_t value) {
-  return ~((uint32_t) 0);
-}
 
-// you must implement this function. It is called when the devie receives a DeviceStatus packet.
-// the return value is the value field of the DeviceRespond packet hibike will respond with
-uint32_t device_status(uint8_t param) {
-  return ~((uint32_t) 0);
+// You must implement this function.
+// It is called when the device receives a Device Write packet.
+// Updates param to new value passed in data.
+//    param   -   Parameter index
+//    data    -   value to write, in little-endian bytes
+//    len     -   number of bytes in data
+//
+//   return  -   size of bytes written on success; otherwise return 0
+
+uint32_t device_write(uint8_t param, uint8_t* data, size_t len){
+  return 0;
 }
 
 
-// you must implement this function. It is called with a buffer and a maximum buffer size.
-// The buffer should be filled with appropriate data for a DataUpdate packer, and the number of bytes
-// added to the buffer should be returned. 
+// You must implement this function.
+// It is called when the device receives a Device Data Update packet.
+// Modifies data_update_buf to contain the parameter value.
+//    param           -   Parameter index
+//    data_update_buf -   buffer to return data in, little-endian
+//    buf_len         -   Maximum length of the buffer
 //
-// You can use the helper function append_buf.
-// append_buf copies the specified amount data into the dst buffer and increments the offset
-uint8_t data_update(uint8_t* data_update_buf, size_t buf_len) {
-  // Read sensor
-  if (buf_len < sizeof(uint16_t) * NUM_PINS) {
+//    return          -   sizeof(param) on success; 0 otherwise
+
+uint8_t device_read(uint8_t param, uint8_t* data, size_t len) {
+  // Read sensor, which is in uint16
+  if (len < sizeof(float) || param > 2) {
```

Comment:
You made a comment on another branch, reminding me that this is a uint, so it can't be negative.  I withdraw this comment.  THanks!